testing/upload_validation.ts

```diff
@@ -1682,6 +1682,14 @@ const legacyPackMetadataSchema = validateFormulas(
 
     (data.syncTables || []).forEach((syncTable, i) => {
       const schema: ObjectSchema<any, any> = syncTable.schema;
+      const propertyNames = Object.keys(schema.properties);
+      for (const dupe of getNonUniqueElements(propertyNames)) {
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          path: ['syncTables', i, 'properties', dupe],
+          message: `Sync table schema property names must be unique. Found duplicate name "${dupe}" in sync table ${syncTable.name}.`,
```

Comment:
You are right. This is not meant to be the only guardrail, though — on the Coda servers, we'll also validate and surface a user-visible error to users whose Pack schemas also violate this uniqueness constraint.

I think it's useful to explicitly tell makers early-on that they can't have `serialNumber` and `sErialNumber` in their sync table schemas, since it's proactive rather than reactive.

Reply:
Can you give me a lot more context?

What will happen to existing sync tables that have already violated this? Are the changes associated with 172, 173, or will they go live immediately? 

My concern is that since Jira fields can be whatever, this will break Jira syncs. I was planning to make the getSchema formula resilient to such name collisions, but that requires the fixedId support, which is currently only supported in 173. Are you planning to just let some Jira syncs break?

Reply:
I wonder if the path of least resistance might be to backport `fixedId`

Reply:
If we wanted to also raise an error for dynamic schemas I think https://github.com/coda/packs-sdk/blob/eb7310fe92969e55fadfee9889b7db00d1b0acbd/testing/upload_validation.ts#L191 is what checks that, but it seems like that'd be blocked on having some kind of support for Jira in the current live schema.

Reply:
Maybe push the implementation of that method into SyncTableGrid v1 & v2?

Reply:
> I wonder if the path of least resistance might be to backport `fixedId`

I talked with Patrick offline on Friday and agree, it seems like `fixedId` is a promising candidate to upgrade v1 syncs to v2. Is that what you were referring to when you say "backporting"?

Reply:
I meant implementing `fixedId` in v1 syncs so modifying the Jira pack to avoid duplicate names in the Coda schema could be done independently of and prior to rolling out the 173 doc schema